### ISSUE TITLE
BUG: image.good_fft_size

### ISSUE THREAD

**Issue by jecampagne** [GalSim-developers/JAX-GalSim#23]
In the code  of `def good_fft_size(cls, input_size):` some lines comes from ` https://github.com/GalSim-developers/GalSim/blob/ece3bd32c1ae6ed771f2b489c5ab1b25729e0ea4/src/Image.cpp#L1009` but the C++ code argument is `input` instead of `input_size` so one should revisit the following lines
```python
        if input <= 2:
            return 2
        # Reduce slightly to eliminate potential rounding errors:
        insize = (1.0 - 1.0e-5) * input
```
with the correct arg naming.

But now there are more pbs looking at this kind of code where `std::` C++ namespace have been replaced by `np.`  
```python
log2n3 = np.max(log2n3, np.log(6.0))  # must be even number
        Nk = int(np.ceil(np.exp(np.min(log2n, log2n3)) - 1.0e-5))
```
`np` is used, but even with numpy this code cannot work `np.min` and `np.max` functions do not act as `min`and `max` as they are dealing with arrays. So this routine should be revisited completly, I guess.

A possible attempt is 
```python
  def good_fft_size(input_size):
        """Round the given input size up to the next higher power of 2 or 3 times a power of 2.

        This rounds up to the next higher value that is either 2^k or 3*2^k.  If you are
        going to be performing FFTs on an image, these will tend to be faster at performing
        the FFT.
        """
        # Reference from GalSim C++
        # https://github.com/GalSim-developers/GalSim/blob/ece3bd32c1ae6ed771f2b489c5ab1b25729e0ea4/src/Image.cpp#L1009
        input_size = int(input_size)
        if input_size <= 2:
            return 2
        # Reduce slightly to eliminate potential rounding errors:
        insize = (1.0 - 1.0e-5) * input_size
        log2n = jnp.log(2.0) * jnp.ceil(np.log(insize) / np.log(2.0))
        log2n3 = jnp.log(3.0) + jnp.log(2.0) * jnp.ceil(
            (jnp.log(insize) - jnp.log(3.0)) / jnp.log(2.0)
        )
        log2n3 = max(log2n3, jnp.log(6.0))  # must be even number
        Nk = int(jnp.ceil(jnp.exp(min(log2n, log2n3)) - 1.0e-5))
        return Nk
```
Now, the gradient of such function is not possible. But is it critical?

**Comment by ismael-mendoza** [comment]
Not critical to get derivatives in this function, probably want to add a documentation saying that you need to set the max fft size with gsparams to actually make drawing jittable/vmappable
